plugins/inputs/jenkins/jenkins.go

```diff
@@ -0,0 +1,483 @@
+
+	fields["response_time"] = info.MonitorData.Hudson_NodeMonitors_ResponseTimeMonitor.Average
+	if diskSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_DiskSpaceMonitor; diskSpaceMonitor != nil {
+		diskSpaceMonitorRoute := "hudson.node_monitors.DiskSpaceMonitor"
+		var diskSpace map[string]interface{}
+		if diskSpace, ok = diskSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(te, diskSpaceMonitor, "map[string]interface{}", diskSpaceMonitorRoute)
+		}
+		if tags["disk_path"], ok = diskSpace["path"].(string); !ok {
+			return badFormatErr(te, diskSpace["path"], "string", diskSpaceMonitorRoute+".path")
+		}
+		if fields["disk_available"], ok = diskSpace["size"].(float64); !ok {
+			return badFormatErr(te, diskSpace["size"], "float64", diskSpaceMonitorRoute+".size")
+		}
+	}
+
+	if tempSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_TemporarySpaceMonitor; tempSpaceMonitor != nil {
+		tempSpaceMonitorRoute := "hudson.node_monitors.TemporarySpaceMonitor"
+		var tempSpace map[string]interface{}
+		if tempSpace, ok = tempSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(te, tempSpaceMonitor, "map[string]interface{}", tempSpaceMonitorRoute)
+		}
+		if tags["temp_path"], ok = tempSpace["path"].(string); !ok {
+			return badFormatErr(te, tempSpace["path"], "string", tempSpaceMonitorRoute+".path")
+		}
+		if fields["temp_available"], ok = tempSpace["size"].(float64); !ok {
+			return badFormatErr(te, tempSpace["size"], "float64", tempSpaceMonitorRoute+".size")
+		}
+	}
+
+	if swapSpaceMonitor := info.MonitorData.Hudson_NodeMonitors_SwapSpaceMonitor; swapSpaceMonitor != nil {
+		swapSpaceMonitorRouter := "hudson.node_monitors.SwapSpaceMonitor"
+		var swapSpace map[string]interface{}
+		if swapSpace, ok = swapSpaceMonitor.(map[string]interface{}); !ok {
+			return badFormatErr(te, swapSpaceMonitor, "map[string]interface{}", swapSpaceMonitorRouter)
+		}
+		if fields["swap_available"], ok = swapSpace["availableSwapSpace"].(float64); !ok {
+			return badFormatErr(te, swapSpace["availableSwapSpace"], "float64", swapSpaceMonitorRouter+".availableSwapSpace")
+		}
+		if fields["swap_total"], ok = swapSpace["totalSwapSpace"].(float64); !ok {
+			return badFormatErr(te, swapSpace["totalSwapSpace"], "float64", swapSpaceMonitorRouter+".totalSwapSpace")
+		}
+		if fields["memory_available"], ok = swapSpace["availablePhysicalMemory"].(float64); !ok {
+			return badFormatErr(te, swapSpace["availablePhysicalMemory"], "float64", swapSpaceMonitorRouter+".availablePhysicalMemory")
+		}
+		if fields["memory_total"], ok = swapSpace["totalPhysicalMemory"].(float64); !ok {
+			return badFormatErr(te, swapSpace["totalPhysicalMemory"], "float64", swapSpaceMonitorRouter+".totalPhysicalMemory")
+		}
+	}
+	return nil
+}
+
+func (j *Jenkins) gatherNodesData(nodes []*gojenkins.Node, acc telegraf.Accumulator) {
+
+	tags := map[string]string{}
+	fields := make(map[string]interface{})
+	baseTe := typedErr{
+		url: j.URL + "/computer/api/json",
```

Comment:
We don't actually know the path because we are using go-jenkins, and while this is the path it uses would subvert the ownership of the library to decide on the paths.  I don't think we should have any url guessing in the plugin code, unless it is possible to have the library tell us the path.

Reply:
Totally make sense, I actually have a question for this. The url should be quite important for debugging. But gojenkins lib won't really return that path. Since I already forked that gojenkins lib, do you think we submit another patch to expose the URL as well?